src/test/coll.rs

```diff
@@ -720,3 +723,34 @@ async fn find_one_and_delete_hint_server_version() {
         assert!(res.is_ok());
     }
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn no_read_preference_to_standalone() {
```

Comment:
We already have the correct behavior required by the drivers ticket, so I implemented a test that verifies it so we don't regress